### ISSUE TITLE
Bug: sam installs devDependencies when using esbuild

### ISSUE THREAD

**Issue by jedwards1211** [aws/aws-sam-cli#4040]
### Description:

I want to use the esbuild preview feature, but it appears to intentionally include `devDependencies` in the build output, and that's too much bloat.

### Steps to reproduce:

Set up a SAM function project that uses `esbuild`, along the lines of https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-build-typescript.html and run `sam build --debug`.

### Observed result:

`sam build` calls `npm install` with `--production=false`:

```
2022-07-07 21:06:28,374 | executing NPM: ['npm', 'install', '-q', '--no-audit', '--no-save', '--production=false', '--unsafe-perm']
```

If I comment out the `BuildMethod: esbuild` line in my `template.yml`, it calls `npm install` with `--production`:

```
2022-07-07 21:08:45,607 | executing NPM: ['npm', 'install', '-q', '--no-audit', '--no-save', '--production', '--unsafe-perm']
```

### Expected result:

The build output never contains `devDependencies`.

It seems like `sam` relies on having `esbuild` get installed in the build directory.  I don't think that approach is going to work in the long run.  Hopefully you can call `esbuild` from the project root directory and output to the build directory, and avoid installing any `devDependencies` in the build directory.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macOS
2. `sam --version`: 1.53.0
3. AWS region: N/A

**Comment by mndeveci** [comment]
Thanks for your feedback. I will be checking with team and we will get back to you with next steps.

**Comment by fabge** [comment]
Any updates here? We have the same issue.

**Comment by mildaniel** [comment]
Hey everyone, this was something we intentionally changed in the GA version of this feature. Please try updating to version 1.56.0 of the SAM-CLI and see if that resolves the issue for you.

**Comment by tunesmith** [comment]
This behavior was fixed for me as of 1.56.0. Note that we now need to install `esbuild` separately. Thank you for the work!!

**Comment by sriram-mv** [comment]
Closing this issue.
